packages/components/src/toolbar-button/index.js

```diff
@@ -12,63 +12,55 @@ import { useContext } from '@wordpress/element';
  * Internal dependencies
  */
 import Button from '../button';
+import ToolbarItem from '../toolbar-item';
 import ToolbarContext from '../toolbar-context';
-import AccessibleToolbarButtonContainer from './accessible-toolbar-button-container';
 import ToolbarButtonContainer from './toolbar-button-container';
 
 function ToolbarButton( {
 	containerClassName,
-	icon,
-	title,
-	shortcut,
-	subscript,
-	onClick,
 	className,
-	isActive,
-	isDisabled,
 	extraProps,
 	children,
+	...props
 } ) {
-	// It'll contain state if `ToolbarButton` is being used within
-	// `<Toolbar __experimentalAccessibilityLabel="label" />`
 	const accessibleToolbarState = useContext( ToolbarContext );
 
-	const button = (
-		<Button
-			icon={ icon }
-			label={ title }
-			shortcut={ shortcut }
-			data-subscript={ subscript }
-			onClick={ ( event ) => {
-				event.stopPropagation();
-				if ( onClick ) {
-					onClick( event );
-				}
-			} }
-			className={ classnames(
-				'components-toolbar__control',
-				className,
-			) }
-			isPressed={ isActive }
-			disabled={ isDisabled }
-			{ ...extraProps }
-		/>
-	);
-
-	if ( accessibleToolbarState ) {
+	if ( ! accessibleToolbarState ) {
+		// This should be deprecated when <Toolbar __experimentalAccessibilityLabel="label">
+		// becomes stable.
 		return (
-			<AccessibleToolbarButtonContainer className={ containerClassName }>
-				{ button }
-			</AccessibleToolbarButtonContainer>
+			<ToolbarButtonContainer className={ containerClassName }>
+				<Button
+					icon={ props.icon }
+					label={ props.title }
+					shortcut={ props.shortcut }
+					data-subscript={ props.subscript }
```

Comment:
I will investigate in a follow-up whether we can get rid of `subscript` as I think we no longer use it in code and it probably was never meant to exist as public API.

Reply:
I kept it there so as to avoid breaking anything, but it's not being used in the new implementation (using `ToolbarItem`) anyway.